content/projects/github-api-consume-part-2/_index.md

```diff
@@ -0,0 +1,58 @@
+---
+content_type: project
+flavours:
+- any_language
+learning_outcomes:
+- web_dev_http_requests
+- web_dev_api_call_tools
+- web_dev_restful_apis
+- web_dev_json_data_representation
+- web_dev_api_consumption
+prerequisites:
+  hard:
+  - projects/github-api-consume-part-1
+  soft: []
+ready: true
+story_points: 3
+submission_type: repo
+tags:
+- pagination
+title: Add pagination to Consume Github API
+---
+
+In part 1 of this project, you made use of Github's awesome API. Now that you are familiar with how it works, you are going to upgrade that project by adding pagination. 
+
+API Pagination is essential if you're dealing with alot of data and endpoints. It helps make your response easier to handle.
+
+## Instructions
+
+In your language of choice(Java, Python, JS) upgrade your existing function to take 2 extra input arguments:
+
+- limit 
+- offset 
+
+This means that if your function worked like so:
+```
+getPullRequests("owner", "repoName", "2022-01-01", "2022-05-05")
```

Comment:
It would be better to make getPullRequests return a complete dataset. 

Update the implementation of `getPullRequests` so that it fetches all the pages.  `getPullRequests` should just work, it needs to be clever enough to fetch all the data it needs to return. 

If this code was used in a larger program then why should the client code care about github's pagination mechanisms? The client should be able to ask for the pull requests and then receive the whole lot.